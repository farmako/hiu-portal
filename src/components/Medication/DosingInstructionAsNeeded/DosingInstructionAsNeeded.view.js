import React from 'react';
import * as PropTypes from 'prop-types';

const DosingInstructionAsNeeded = (props) => {
  const { dosage } = props;
  if (dosage.asNeededBoolean) {
    return (
      <li>
        <span>Take as needed</span>
      </li>
    );
  }
  return null;
};

DosingInstructionAsNeeded.propTypes = {
  dosage: PropTypes.shape({
    asNeededBoolean: PropTypes.bool,
  }),
};

DosingInstructionAsNeeded.defaultProps = {
  dosage: {
    asNeededBoolean: false,
  },
};

export default DosingInstructionAsNeeded;
