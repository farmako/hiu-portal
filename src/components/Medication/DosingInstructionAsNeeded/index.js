import DosingInstructionAsNeeded from './DosingInstructionAsNeeded.view';

export default DosingInstructionAsNeeded;
