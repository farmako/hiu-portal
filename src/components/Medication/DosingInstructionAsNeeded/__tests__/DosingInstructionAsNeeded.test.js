import React from 'react';
import { shallow } from 'enzyme';
import DosingInstructionAsNeeded from '../DosingInstructionAsNeeded.view';

describe('DosingInstructionAsNeeded', () => {
  const mockData = {
    sequence: 1,
    text: '22/01/2020 to 07/02/2020',
    timing: {
      event: ['22/1/2019'],
    },
    asNeededBoolean: true,
  };
  const wrapper = shallow(
    <DosingInstructionAsNeeded dosage={mockData} />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render data as needed', () => {
    expect(wrapper.find('span').text()).toEqual('Take as needed');
  });

  it('should not render anything if asNeededBoolean is provided as false', () => {
    mockData.asNeededBoolean = false;
    wrapper.setProps({ dosage: mockData });
    expect(wrapper.find('li').at(0).exists()).toEqual(false);
  });
});
