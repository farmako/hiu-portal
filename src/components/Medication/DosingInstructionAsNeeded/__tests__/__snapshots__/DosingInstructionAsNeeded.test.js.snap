// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DosingInstructionAsNeeded should render properly 1`] = `
<li>
  <span>
    Take as needed
  </span>
</li>
`;
